### PR TITLE
XW-4371 Add ads to JW player generated from WikiText tags

### PR DESCRIPTION
**Pull request by nandy-andy**
https://wikia-inc.atlassian.net/browse/ADEN-6371

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js [hunk 1]
```diff
@@ -4,9 +4,11 @@ define('wikia.articleVideo.featuredVideo.ads', [
 	'ext.wikia.adEngine.slot.service.megaAdUnitBuilder',
 	'ext.wikia.adEngine.slot.service.srcProvider',
 	'ext.wikia.adEngine.video.vastDebugger',
-	'wikia.articleVideo.featuredVideo.adsTracking',
+	'ext.wikia.adEngine.video.player.jwplayer.adsTracking',
```
Comment (fraszczakszymon) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js]: Keep alphabetical order please

File: extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			player.on('beforePlay', function () {
+				if (prerollPositionReached) {
+					return;
+				}
+				log('Preroll position reached', log.levels.info, logGroup);
+
+				correlator = Math.round(Math.random() * 10000000000);
+				videoDepth += 1;
+
+				trackingParams.adProduct = 'in-article-video-preroll';
```
Comment (fraszczakszymon) [inline extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js]: We should reset this value to `video` on `beforeComplete`. Like here: https://github.com/Wikia/app/blob/dev/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.ads.js#L119

File: extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		return vastUrlBuilder.build(aspectRatio, slotParams, options);
+	}
+
+	return function(player) {
+		var correlator,
+			videoElement = player && player.getContainer && player.getContainer(),
+			videoContainer = videoElement && videoElement.parentNode,
+			prerollPositionReached = false,
+			trackingParams = {
+				adProduct: 'in-article-video',
```
Comment (fraszczakszymon) [inline extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js]: Can we set just `video` here? So it'll be consistent with N&S: https://github.com/Wikia/f2/blob/master/frontend/components/ads/slots.js#L147

Comment (nandy-andy) [reply extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js]: Sure :)

File: extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js (hunk 2 above)
Comment (nandy-andy) [reply extensions/wikia/JWPlayerTag/scripts/jwplayertag.ads.js]: `beforeComplete` will be used for post-rolls if we need them. Let's not reset anything or if we find a bug reset on `complate` maybe? ;)

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**